### PR TITLE
Add CI via GitHub Actions

### PR DESCRIPTION
**Pull request by jcharum**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go: [1.12, 1.13]
```
Comment (siddharthab) [inline .github/workflows/ci.yml:13]: Maybe test 1.14 as well?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: CI
```
Comment (siddharthab) [inline .github/workflows/ci.yml:1]: opt: Maybe call it Tests?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
```
Comment (siddharthab) [inline .github/workflows/ci.yml:11]: You can set `fail-fast: false` to continue one release even if the other fails.

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    name: Build & Test
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Go ${{ matrix.go }}
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go }}
+    - name: Check out
+      uses: actions/checkout@v2
+    - name: Build
```
Comment (siddharthab) [inline .github/workflows/ci.yml:23]: I don't think you need a separate build step. `test` will also build all packages even if they don't have any `_test.go` file in them.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      run: go build -v ./...
+    - name: Test
+      run: go test -v -short ./...
```
Comment (siddharthab) [inline .github/workflows/ci.yml:26]: opt: Maybe not have the `-v` flag to reduce log noise.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (jcharum) [reply .github/workflows/ci.yml]: Removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jcharum) [reply .github/workflows/ci.yml]: Sounds good.

**Comment by siddharthab** [comment]
Just noticed that this was still a draft. The email does not make it clear. Sorry for the unsolicited review.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (jcharum) [reply .github/workflows/ci.yml]: Good idea.

**Comment by jcharum** [comment]
> Just noticed that this was still a draft. The email does not make it clear. Sorry for the unsolicited review.

@siddharthab: No worries! Sorry for wasting your time with a not-quite-done thing. I feel like drafts should basically be invisible to everyone but the author -- sending notifications seems like the opposite of that.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jcharum) [reply .github/workflows/ci.yml:1]: I'm also running a linter, so I think its general intent is "stuff that we do on branches to support continuous integration".

File: .github/workflows/ci.yml (hunk 3 above)
Comment (jcharum) [reply .github/workflows/ci.yml:11]: Good idea. Done.

**Comment by jcharum** [comment]
@siddharthab: I think it's actually good to go now. Let me know if you have any other comments/suggestions. Thanks!

**Comment by siddharthab** [comment]
All good. Merge at will.

**Pull request merged**